account_asset_management/models/account_asset_line.py

```diff
@@ -0,0 +1,279 @@
+# Copyright 2009-2018 Noviat
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+import datetime
+
+from odoo import api, fields, models, _
+import odoo.addons.decimal_precision as dp
+from odoo.exceptions import UserError
+
+
+class AccountAssetLine(models.Model):
+    _name = 'account.asset.line'
+    _description = 'Asset depreciation table line'
+    _order = 'type, line_date'
+
+    name = fields.Char(string='Depreciation Name', size=64, readonly=True)
+    asset_id = fields.Many2one(
+        comodel_name='account.asset', string='Asset',
+        required=True, ondelete='cascade')
+    previous_id = fields.Many2one(
+        comodel_name='account.asset.line',
+        string='Previous Depreciation Line',
+        readonly=True)
+    parent_state = fields.Selection(
+        selection=[
+            ('draft', 'Draft'),
+            ('open', 'Running'),
+            ('close', 'Close'),
+            ('removed', 'Removed')],
+        related='asset_id.state',
+        string='State of Asset',
+        readonly=True,
+    )
+    depreciation_base = fields.Float(
+        related='asset_id.depreciation_base',
+        string='Depreciation Base',
+        readonly=True,
+    )
+    amount = fields.Float(
+        string='Amount', digits=dp.get_precision('Account'),
+        required=True)
+    remaining_value = fields.Float(
+        compute='_compute_values',
+        digits=dp.get_precision('Account'),
+        string='Next Period Depreciation',
+        store=True)
+    depreciated_value = fields.Float(
+        compute='_compute_values',
+        digits=dp.get_precision('Account'),
+        string='Amount Already Depreciated',
+        store=True)
+    line_date = fields.Date(string='Date', required=True)
+    move_id = fields.Many2one(
+        comodel_name='account.move',
+        string='Depreciation Entry', readonly=True)
+    move_check = fields.Boolean(
+        compute='_compute_move_check',
+        string='Posted',
+        store=True)
+    type = fields.Selection(
+        selection=[
+            ('create', 'Depreciation Base'),
+            ('depreciate', 'Depreciation'),
+            ('remove', 'Asset Removal')],
+        readonly=True, default='depreciate')
+    init_entry = fields.Boolean(
+        string='Initial Balance Entry',
+        help="Set this flag for entries of previous fiscal years "
+             "for which OpenERP has not generated accounting entries.")
+
+    @api.depends('amount', 'previous_id', 'type')
+    @api.multi
+    def _compute_values(self):
+        dlines = self
+        if self.env.context.get('no_compute_asset_line_ids'):
+            # skip compute for lines in unlink
+            exclude_ids = self.env.context['no_compute_asset_line_ids']
+            dlines = dlines.filtered(lambda l: l.id not in exclude_ids)
+        dlines = self.filtered(lambda l: l.type == 'depreciate')
```

Comment:
I guess here it should be:

```
dlines = dlines.filtered(lambda l: l.type == 'depreciate')
```

Reply:
since dlines = self it's the same but indeed your suggestion is cleaner.